src/Util/FormViewIterator.php

```diff
@@ -29,7 +29,9 @@ class FormViewIterator implements \RecursiveIterator
 
     public function __construct(FormView $formView)
     {
-        $this->iterator = $formView->getIterator();
+        $iterator = $formView->getIterator();
+        \assert($iterator instanceof \ArrayIterator);
```

Comment:
Are you adding this because the typehint was '\ArrayIterator|Formview[]' ?

We could make a PR to symfony to change | to &.

Reply:
Or maybe we should change our typehint on the iterator property?

Reply:
Yes, I'm adding this because of typehint.

Can you please help me with this? I'm not sure that it will be merged in 4.4.x branch and the release will fast enough on their side (I'm worried that this PR will be stale at that moment)

Reply:
If you do:

```
/** @var \ArrayIterator */
$iterator = $formView->getIterator();
```

It does not work?

Reply:
@vladyslavstartsev I dont consider the Symfony should be a blocking point, we could just open the PR on symfony and add the link here.

@jordisala1991 phpstan recommend the use of assert over phpdoc

Reply:
I'm leaving `assert`  because of

> phpstan recommend the use of assert over phpdoc

Reply:
the PR

https://github.com/symfony/symfony/pull/37760